### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Ruchit-rt**
testing if this works. made small changes

### CODE REVIEW

File: parser.py [hunk 1]
```diff
@@ -22,52 +30,28 @@ def headlines(design = "APEX"):
     try:
```
Comment (Akshat-Tripathi) [inline parser.py:30]: try to get rid of this

File: parser.py [hunk 2]
```diff
@@ -9,7 +10,14 @@ def headlines(design = "APEX"):
 
     # split into physical attributes and simulation results
     rocket = root[0]
-    sims = root[1]
+    sims   = root[1]
+
+    def simdata(rootObj,dataset,att):
```
Comment (Akshat-Tripathi) [inline parser.py:15]: I think you forgot to return the result here

File: parser.py [hunk 3]
```diff
@@ -22,52 +30,28 @@ def headlines(design = "APEX"):
...
-
-    summary += f"**Motor selected for use:** {motor} <br/> \n"
-    summary += f"**Apogee:** {apogee} m <br/> \n"
-    summary += f"**Max speed:** {maxvel} m/s <br/> \n"
-    summary += f"**Max acceleration:** {maxacc} m/s^2 <br/> \n"
-    summary += f"**Flight duration:** {flightdur} s <br/> \n"
-    summary += f"**Ground hit velocity:** {groundhit} m/s <br/> \n"
-    summary += f"**Dry mass:** {mass} kg "
+    sumlist = [f"**Motor selected for use:** {motor} <br/> \n", f"**Apogee:** {apogee} m <br/> \n", f"**Max speed:** {maxvel} m/s <br/> \n", f"**Max acceleration:** {maxacc} m/s^2 <br/> \n", f"**Flight duration:** {flightdur} s <br/> \n", f"**Ground hit velocity:** {groundhit} m/s <br/> \n", f"**Dry mass:** {mass} kg "]
+    summary = sumlist.join('')
```
Comment (Akshat-Tripathi) [inline parser.py:54]: I like the functional programming, but the other one was cleaner rip

File: parser.py [hunk 4]
```diff
@@ -22,52 +30,28 @@ def headlines(design = "APEX"):
     try:
         mass = 0
         for element in rocket.iter('overridemass'):
-            mass = mass + float(element.text)
+            mass += float(element.text)
         for element in rocket.iter('mass'):
-            mass = mass + float(element.text)
+            mass += float(element.text)
```
Comment (Akshat-Tripathi) [inline parser.py:35]: yay

File: parser.py [hunk 5]
```diff
@@ -1,3 +1,4 @@
+from abc import abstractproperty
```
Comment (Akshat-Tripathi) [inline parser.py:1]: what's this for?

File: parser.py [hunk 6]
```diff
@@ -22,52 +30,28 @@ def headlines(design = "APEX"):
...
-    try:
-        flightdur = sims[0].find('flightdata').attrib['flighttime']
-    except:
-        flightdur = "Error parsing simulation data"
-
-    try:
-        groundhit = sims[0].find('flightdata').attrib['groundhitvelocity']
-    except:
-        groundhit = "Error parsing simulation data"
+    apogee    = simdata (sims,'flightdata','maxaltitude')
```
Comment (Akshat-Tripathi) [inline parser.py:41]: Looks good

**Review by Akshat-Tripathi — CHANGES_REQUESTED** [review]

File: parser.py (hunk 5 above)
Comment (Ruchit-rt) [reply parser.py:1]: I don't know how this got here really :). Removed this now

File: parser.py (hunk 2 above)
Comment (Ruchit-rt) [reply parser.py:15]: done

File: parser.py (hunk 3 above)
Comment (shreybohra) [reply parser.py:61]: can we keep the legible version lol i want to be able to edit this without having to use duolingo

Comment (Akshat-Tripathi) [reply parser.py:52]: yeah we should keep it clean

File: parser.py [hunk 7]
```diff
@@ -9,66 +9,47 @@ def headlines(design = "APEX"):
...
+        return res 
+    
     # get motor
-    try:
-        for element in rocket.iter('motorconfiguration'):
-            motor = element[0].text
-    except:
-        motor = "No motor found"
+    motor   = "motor not found"
+    element = rocket.iter('motorconfiguration')
```
Comment (Akshat-Tripathi) [inline parser.py:23]: Maybe this should be a for loop? @shreybohra

**Review by Akshat-Tripathi — CHANGES_REQUESTED** [review]
Since this is gonna be a git hook, let's remove the create_commit function and turn this into a cli tool. Then the hook can do smth like: python3 main.py APEX

File: parser.py (hunk 7 above)
Comment (shreybohra) [reply parser.py:23]: yeah iter method needs to be used in a for loop as far as I know

File: parser.py [hunk 8]
```diff
@@ -48,11 +50,13 @@ def simdata(rootObj,dataset,att):
     # see raw xml to get attrib keys
 
     # pretty print data
-    sumlist = [f"**Motor selected for use:** {motor} <br/> \n", f"**Apogee:** {apogee} m <br/> \n", f"**Max speed:** {maxvel} m/s <br/> \n", f"**Max acceleration:** {maxacc} m/s^2 <br/> \n", f"**Flight duration:** {flightdur} s <br/> \n", f"**Ground hit velocity:** {groundhit} m/s <br/> \n", f"**Dry mass:** {mass} kg "]
-    summary = sumlist.join('')
+    # sumlist = [f"**Motor selected for use:** {motor} <br/> \n", f"**Apogee:** {apogee} m <br/> \n", f"**Max speed:** {maxvel} m/s <br/> \n", f"**Max acceleration:** {maxacc} m/s^2 <br/> \n", f"**Flight duration:** {flightdur} s <br/> \n", f"**Ground hit velocity:** {groundhit} m/s <br/> \n", f"**Dry mass:** {mass} kg "]
+    # summary = sumlist.join('')
+    sumlist = "**Motor selected for use:** {0} <br/> \n **Apogee:** {1} m <br/> \n **Max speed:** {2} m/s <br/> \n **Max acceleration:** {3} m/s^2 <br/> \n **Flight duration:** {4} <br/> \n **Ground hit velocity:** {5} m/s <br/> \n **Dry mass:** {6} kg "
```
Comment (shreybohra) [inline parser.py:55]: can we keep the original summary structure? it's just nicer to read

**Review by shreybohra — CHANGES_REQUESTED** [review]

**Pull request closed without merging**